frontend/src/components/validation-window/validation-window.component.tsx

```diff
@@ -1,5 +1,36 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+import { useSelector } from "react-redux";
+
+import Alert from "@material-ui/lab/Alert";
+import { ApplicationStateModel } from "../../state/models/application-state.model";
+import { ScheduleErrorMessageModel } from "../../state/models/schedule-data/schedule-error-message.model";
 
 export function ValidationWindowComponent() {
-  return <div>"ValidationWindowComponent works !"</div>;
+  //#region members
+  const [errors, setErrors] = useState<ScheduleErrorMessageModel[]>();
+  //#endregion
+
+  //#region state interaction
+  const errors_received = useSelector((state: ApplicationStateModel) => state.scheduleErrors);
+
+  useEffect(() => {
+    if (errors_received) {
+      setErrors(errors_received);
+    }
+  }, [errors_received]);
+
+  //#endregion
+
+  //#region view
+
+  function alert(message) {
+    return (
+      <Alert variant="outlined" severity="warning">
+        {message}
+      </Alert>
+    );
+  }
+
+  return <div>{errors?.map((r) => alert(r.message))}</div>;
```

Comment:
Nie wiem, czy użycie osobnej funkcji w tym miejscu ma duży sens, więc ja bym ja usunął. 

Ogólnie, to nie jest dobrze dodawać elementy do DOM w taki sposób jak ty dodajesz tu. Nie wiem dlaczego, ale kiedy `functional component` zostaje utworzony w taki sposób, to w nim nie dzialają hooki, a `React` wyrzuca taki błąd 
```
Rendered fewer hooks than expected
```

Zamiast tego, lepiej użyć innej składni:
```
<FunctionName prop1={...} prop2={...}
```
Wtedy wszystko jest dobrze. 

Podsyłam też artykuł który opisuje ten problem:

https://medium.com/@jonchurch/how-to-fix-react-error-rendered-fewer-hooks-than-expected-e6a378985d3c

Tu pewnie nie będzie problemu, ale lepiej o tym wiedzieć jeżeli w przyszłości będziesz chciał dodać do takiego komponenta hook

Reply:
Jeszcze jedna uwaga, to to że zawsze kiedy generujesz listę w reakcie - musisz ustawiać klucz, ponieważ na podstawie tego framework wyznacza czy to jest nowy element i czy warto go renderować ponownie

https://reactjs.org/docs/reconciliation.html#recursing-on-children


Tu jest też przykład na CodePen, który demonstruje co może być gdy nie ma klucza( lub gdy on nie jest unikatowy dla elementa):
https://codepen.io/pen?&editable=true&editors=0010

Jeżeli wpiszesz w pierwszy input jakiś tekst, a potem klikniesz add new to start, to zobaczysz, że tekst pozostał w pierwszym inpucie, mimo tego, że musiałby by być w drugim. ( w tym przykladzie indeks był uzyty jako klucz, i react zapamiętał, że ty wpisywałeś wartość do obiektu o kluczu 0. I z tego powodu, kiedy dodałeś nowy element na początku, to w nim pojawił się tekst, zamiast być w elemencie z klucem 1)